### PR TITLE
Support React Native 0.71

### PR DESCRIPTION
**Pull request by j-piasecki**
## Description

This PR adds support for new architecture on React Native 0.71, which most notably includes the usage of prefabs in native builds.

It also removes most of the dependencies of `FabricExample` app, so upgrades in the future should be easier.

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -198,11 +202,7 @@ def kotlin_version = safeExtGet('kotlinVersion', project.properties['RNGH_kotlin
 
 dependencies {
     //noinspection GradleDynamicVersion
-    if (isNewArchitectureEnabled()) {
-        implementation project(':ReactAndroid')
-    } else {
-        implementation 'com.facebook.react:react-native:+'
-    }
+    implementation 'com.facebook.react:react-native:+' // TODO
```
Comment (tomekzaw) [inline android/build.gradle:205]:
```suggestion
    if (REACT_NATIVE_MINOR_VERSION >= 71) {
         implementation "com.facebook.react:react-android" // version substituted by RNGP
    } else {
        implementation "com.facebook.react:react-native:+" // From node_modules
    }
```

Comment (j-piasecki) [reply android/build.gradle]: Done in https://github.com/software-mansion/react-native-gesture-handler/pull/2313/commits/ae8149a057b300982906209470e214ecfd6c8e4e

File: FabricExample/patches/react-native+0.71.0-rc.0.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake b/node_modules/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake
```
Comment (tomekzaw) [inline FabricExample/patches/react-native+0.71.0-rc.0.patch:1]: plz remove this file

File: FabricExample/package.json [hunk 3]
```diff
@@ -5,46 +5,40 @@
...
+    "metro-react-native-babel-preset": "0.73.5",
+    "prettier": "^2.4.1",
+    "react-test-renderer": "18.2.0",
+    "typescript": "4.8.4"
   },
   "jest": {
     "preset": "react-native"
+  },
+  "resolutions": {
+    "@react-native-community/cli-platform-android": "10.0.0-alpha.4"
```
Comment (tomekzaw) [inline FabricExample/package.json:42]: no longer necessary, don't forget to update yarn.lock

**Review by tomekzaw — CHANGES_REQUESTED** [review]

File: FabricExample/package.json (hunk 3 above)
Comment (j-piasecki) [reply FabricExample/package.json]: Done in https://github.com/software-mansion/react-native-gesture-handler/pull/2313/commits/f6df763638b1b41cd2aaa1cf08d2b396e55676e3.

**Review by tomekzaw — APPROVED** [review]

File: FabricExample/patches/react-native+0.71.0-rc.0.patch (hunk 2 above)
Comment (j-piasecki) [reply FabricExample/patches/react-native+0.71.0-rc.0.patch]: Done in https://github.com/software-mansion/react-native-gesture-handler/pull/2313/commits/f6df763638b1b41cd2aaa1cf08d2b396e55676e3.

File: android/build.gradle [hunk 4]
```diff
@@ -27,25 +27,33 @@ def isNewArchitectureEnabled() {
...
+        return reactNativeFromProjectNodeModules
+    }
+
+    def reactNativeFromNodeModulesWithReanimated = file("${projectDir}/../../react-native")
+    if (reactNativeFromNodeModulesWithReanimated.exists()) {
+        return reactNativeFromNodeModulesWithReanimated
+    }
+
+    throw new Exception(
+        "[react-native-gesture-native] Unable to resolve react-native location in " +
```
Comment (kkafar) [inline android/build.gradle:53]: Just noticed a typo:

```suggestion
        "[react-native-gesture-handler] Unable to resolve react-native location in " +
```

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**